src/sdk/analytics/utils/navigation.ts

```diff
@@ -0,0 +1,185 @@
+const THIRTY_MINUTES_IN_MS = 1_800_000
+const ONE_YEAR_IN_MS = 31_536_000_000
+const SESSION_ID = 'VtexRCSessionIdv7'
+const MAC_ID = 'VtexRCMacIdv7'
+
+/**
+ * Generates an Universally Unique Identifier (UUID) that is a 128-bit number, with the encoding in the
+ * format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx. For this, the algorithm is based on version 4 of RFC4122,
+ * which can be better understood in the following documentation: https://www.ietf.org/rfc/rfc4122.txt.
+ *
+ */
+function uuidv4() {
+  const uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+    const r = (Math.random() * 16) | 0
+
+    return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16)
+  })
+
+  return uuid
```

Comment:
To stop SonarQube warning:
```suggestion
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
    const r = (Math.random() * 16) | 0

    return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16)
  })
```